front-end/stateofthecards-front-end/src/components/QuickJoinMenu.tsx

```diff
@@ -0,0 +1,144 @@
+import React from "react";
+import styles from "./QuickJoinMenu.module.css";
+import ScaleLoader from "react-spinners/ScaleLoader";
+
+enum QuickJoinState {
+	Idle,
+	Searching,
+	InLobby,
+}
+
+interface IProps {
+	onClickCancel: () => void;
+}
+
+interface IState {
+	currentState: QuickJoinState;
+	lobbyInfo?: LobbyInfo;
+}
+
+interface LobbyInfo {
+	// The current players connected to the lobby
+	//  Should not use string-type in the future.
+	//  so that a player can press on a different,
+	//  player to show info.
+	players: string[];
+
+	// The id of the game so it can
+	//  be fetched by the user.
+	gameInfo: GameInfo;
+
+	// Should be an enum similar to:
+	//  Waiting, Preparing, Post
+	state: string;
```

Comment:
If this "Should be an enum", why is it not?

Reply:
Because we are not that far in yet.

Reply:
Since the back-end has not been implemented yet, A decision has to be made first on how we are going to implement this. It is only a place-holder for now.